flink-runtime/src/test/java/org/apache/flink/runtime/metrics/groups/MetricGroupTest.java

```diff
@@ -216,6 +219,28 @@ public void testNameCollisionAfterKeyValueGroup() {
 		assertFalse("Value is present in logical scope.", logicalScope.contains(value));
 	}
 
+	/**
+	 * Verifies that calling {@link AbstractMetricGroup#getLogicalScope(CharacterFilter, char, int)} on {@link GenericValueMetricGroup}
+	 * should ignore value as well.
+	 */
+	@Test
+	public void testLogicalScopeShouldIgnoreValueGroupName() {
+		Configuration config = new Configuration();
+		config.setString(ConfigConstants.METRICS_REPORTER_PREFIX + "test." + ConfigConstants.METRICS_REPORTER_CLASS_SUFFIX, TestReporter.class.getName());
+		MetricRegistryImpl registry = new MetricRegistryImpl(MetricRegistryConfiguration.fromConfiguration(config));
```

Comment:
should  be wrapped in a try-catch block

Reply:
Do you mean `registry` should be shutdown in `finally` clause?

Reply:
correct